main.go

```diff
@@ -189,6 +207,7 @@ func runManager(ctx context.Context, mgr manager.Manager, config *managerConfig)
 
 	clusterControllerOpts := []controllers.ControllerConfigOpts{
 		controllers.WithMaxConcurrentReconciles(config.concurrentReconcilesNutanixCluster),
+		controllers.WithRateLimiter(config.rateLimiter),
```

Comment:
https://release-1-5.cluster-api.sigs.k8s.io/developer/architecture/controllers/tuning. will this also impact api server calls?